### PR TITLE
[Fix] Fix CI error due to `np.int` and legacy builder.py

### PR DESCRIPTION
**Pull request by RangeKing**
## Motivation

Fix CI error due to `np.int` and legacy builder.py, https://github.com/open-mmlab/mmdetection/pull/9479

## Modification

1. Replace all `np.int` with `np.int64`.
2. Replace all `build_detector` with `MODELS.build`.
3. Using the `main` branch of mmengine instead of the tag version in CI.

### CODE REVIEW

**Review by PeterH0323 — APPROVED** [review]
LGTM

**Comment by PeterH0323** [comment]
中英文档 `yolov5_description.md` 还有一些，也请一并修改

**Review by hhaAndroid — APPROVED** [review]

File: mmyolo/utils/boxam_utils.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import torchvision
 from mmcv.transforms import Compose
 from mmdet.evaluation import get_classes
-from mmdet.models import build_detector
+from mmdet.registry import MODELS
```
Comment (hhaAndroid) [inline mmyolo/utils/boxam_utils.py:15]: mmyolo.registry

File: tests/test_models/test_detectors/test_yolo_detector.py [hunk 2]
```diff
@@ -29,8 +29,8 @@ def test_init(self, cfg_file):
         model = get_detector_cfg(cfg_file)
         model.backbone.init_cfg = None
 
-        from mmdet.models import build_detector
-        detector = build_detector(model)
+        from mmdet.registry import MODELS
```
Comment (hhaAndroid) [inline tests/test_models/test_detectors/test_yolo_detector.py:32]: mmyolo.

File: tests/test_models/test_detectors/test_yolo_detector.py [hunk 3]
```diff
@@ -56,11 +56,11 @@ def test_forward_loss_mode(self, cfg_file, devices):
                 std=[255., 255., 255.],
                 bgr_to_rgb=True)
 
-        from mmdet.models import build_detector
+        from mmdet.registry import MODELS
```
Comment (hhaAndroid) [inline tests/test_models/test_detectors/test_yolo_detector.py:59]: same

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**